sympy/sets/tests/test_fancysets.py

```diff
@@ -309,6 +309,10 @@ def test_Range_set():
         assert Range(builtin_range(1000000000000)) == \
             Range(1000000000000)
 
+    # test Range.as_relational
+    assert Range(1, 4).as_relational(x) == (x >= 1) & (x <= 3) & Eq(x - 1, floor(x) - 1)
+    assert Range(1, oo).as_relational(x) == (x >= 1) & (x < oo) & Eq(x - 1, floor(x) - 1)
```

Comment:
```suggestion
    assert Range(oo, 1, -2).as_relational(x) == (x >= 3) & (x < oo) & Eq((3 - x)/2, floor((3 - x)/2))
```
This will test the reversal and step requirement.